javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,6 +56,24 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
+                                                             : asArray(iterable.iterator(), size);
+        return ofAll(array, size, 0);
+    }
+
+    static <T> BitMappedTrie<T> ofAll(Object[] array, int size, int depthShift) {
+        if (size == 0) { return empty(); }
+
+        int shift = depthShift;
+        for (; array.length > branchingFactor(); shift += BRANCHING_BASE) {
```

Comment:
Ok, your call.

Mutable static fields + concurrency = 💣 Are we sure our benchmarks / tests are correct?